### PR TITLE
fixes for better handling of times on the edit form

### PR DESCRIPTION
**Pull request by aaronoppenheimer**
Decouples the gig object from the rendering of the form for clearer handling of end dates and better toggling between all-day and not-all-day events.

### CODE REVIEW

File: gig/views.py [hunk 1]
```diff
@@ -146,6 +146,24 @@ def form_valid(self, form):
         return result
 
 
+def SetupFormTimes(context,object):
```
Comment (bklang) [inline gig/views.py:149]: Small nit, but method names typically are underscored, while classes are initial-caps
```suggestion
def setup_form_times(context,object):
```

File: gig/views.py [hunk 2]
```diff
@@ -158,6 +176,9 @@ def test_func(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['timezone'] = self.object.band.timezone
+
+        SetupFormTimes(context,self.object)
```
Comment (bklang) [inline gig/views.py:180]:
```suggestion
        setup_form_times(context,self.object)
```

File: gig/views.py [hunk 3]
```diff
@@ -166,14 +187,9 @@ def get_form_kwargs(self, *args, **kwargs):
...
-        
-        # if this is not a full day event, or the date and enddate are the same,
-        # make the enddate widget on the form blank
-        if not self.object.is_full_day or \
-            self.object.enddate and self.object.date.date() == self.object.enddate.date():
-            kwargs['initial']['enddate'] = None
-        
+        # XXX
+        # kwargs['initial'] = forms.models.model_to_dict(self.object)
+
```
Comment (bklang) [inline gig/views.py:192]:
```suggestion
```

File: gig/views.py [hunk 4]
```diff
@@ -195,28 +211,40 @@ def form_valid(self, form):
...
         if not self.request.user.is_authenticated:
             return self.handle_no_permission()
         gig = get_object_or_404(Gig, id=self.kwargs['pk'])
         return gig.band.is_editor(self.request.user)
 
+    def get_context_data(self, **kwargs):
+        self.original_gig = Gig.objects.get(id=self.kwargs['pk'])
+        context = super().get_context_data(**kwargs)
+
+        SetupFormTimes(context, self.original_gig)
```
Comment (bklang) [inline gig/views.py:230]:
```suggestion
        setup_form_times(context, self.original_gig)
```

File: gig/views.py [hunk 5]
```diff
@@ -195,28 +211,40 @@ def form_valid(self, form):
...
-        kwargs['initial'] = forms.models.model_to_dict(gig_orig)
+        kwargs['initial'] = forms.models.model_to_dict(self.original_gig)
         # ...but replace the title with a 'copy of'
         kwargs['initial']['title'] = f'Copy of {kwargs["initial"]["title"]}'
         return kwargs
 
     def get_band_from_kwargs(self, **kwargs):
-        # didn't have the band from the request args, so pull it from the gig
-        return get_object_or_404(Gig, id=self.kwargs['pk']).band
+        return self.original_gig.band
```
Comment (bklang) [inline gig/views.py:247]: It seems like this method's behavior no longer matches its name. Do we even still need it?

Comment (aaronoppenheimer) [reply gig/views.py:244]: We do need it but the name could be better. Will fix.

**Review by bklang — APPROVED** [review]
Looks good, tests good. Thanks for tackling this!

**Pull request merged**